vc-worker.py

```diff
@@ -279,29 +279,30 @@ def process_other_actions(conn, action, vc):
                 logger.info('request aborted, cannot be done on a machine in such a state')
                 return
 
-        if request_type == 'undeploy':
+        if request_type is RequestType.UNDEPLOY:
             new_machine_state = action_undeploy(request, machine_ro, vc)
-        elif request_type == 'start':
+        elif request_type is RequestType.START:
             new_machine_state = action_start(request, machine_ro, vc)
-        elif request_type == 'stop':
+        elif request_type is RequestType.STOP:
             new_machine_state = action_stop(request, machine_ro, vc)
-        elif request_type == 'get_info':
+        elif request_type is RequestType.GET_INFO:
             action_get_info(request, machine_ro, vc, action, conn)
             return
-        elif request_type == 'take_screenshot':
+        elif request_type is RequestType.TAKE_SCREENSHOT:
             new_machine_state = action_take_screenshot(request, machine_ro, vc, conn)
-        elif request_type == 'take_snapshot':
+        elif request_type is RequestType.TAKE_SNAPSHOT:
             new_machine_state = action_take_snapshot(request, machine_ro, vc, conn)
-        elif request_type == 'restore_snapshot':
+        elif request_type is RequestType.RESTORE_SNAPSHOT:
             new_machine_state = action_restore_snapshot(request, machine_ro, vc, conn)
-        elif request_type == 'delete_snapshot':
+        elif request_type is RequestType.DELETE_SNAPSHOT:
             new_machine_state = action_delete_snapshot(request, machine_ro, vc, conn)
         else:
-            logger.warn('unknown request: {} is going to succeed'.format(request_type))
+            # this should not happen
+            settings.raven.captureMessage(f'Unhandled request type: {request_type}')
             # will not be actually saved, only for setting request as failed
             new_machine_state = MachineState.FAILED
 
-        if request_type in ['undeploy', 'start', 'stop']:  # only these requests can change machine state
+        if request_type.can_change_machine_state():
```

Comment:
also 'deploy' request can change machine state, but i think it does not affect this condition

Reply:
I agree, I also presume it to be correct.

Reply:
in my opinion, it would be more readable to have it: `request_type in [RequestType.STOP, RequestType.DEPLOY, RequestType.UNDEPLOY]:` what do you think?